src/main/java/com/eternalcode/discordapp/filter/FilterMessageEmbedController.java

```diff
@@ -14,7 +14,22 @@ public FilterMessageEmbedController(FilterService filterService) {
     @Override
     public void onMessageReceived(MessageReceivedEvent event) {
         event.getMessage().getEmbeds().forEach(embed -> {
-            FilterResult result = this.filterService.check(embed.getAuthor().getName(), embed.getTitle());
+            if (embed == null) {
+                return;
+            }
+
+            if (embed.getAuthor() == null) {
+                return;
+            }
+
+            String name = embed.getAuthor().getName();
+            String title = embed.getTitle();
+
+            if (name == null || title == null) {
+                return;
+            }
```

Comment:
Można też sprawdzić `isEmpty()` lub `isBlank()`, tak na wszelki wypadek jakby name wynosił pusty string z jakiegoś powodu 🤷🏻

Reply:
Zbędne, jak będzie puste to po prostu result zwróci false